nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/ProcessExtensions.cs

```diff
@@ -37,8 +39,20 @@ public static Task<(int ExitCode, string Output, string Error)> RunAsync(string
 
         process.Exited += (sender, args) =>
         {
-            tcs.TrySetResult((process.ExitCode, stdout.ToString(), stderr.ToString()));
-            process.Dispose();
+            // We must call WaitForExit to make sure we've received all OutputDataReceived/ErrorDataReceived calls
```

Comment:
Expand comment to give context about why using the reset event is necessary.

```suggestion
            // It is necessary to wait until we have invoked 'BeginXReadLine' for our redirected IO. Then,
            // we must call WaitForExit to make sure we've received all OutputDataReceived/ErrorDataReceived calls
```